### ISSUE TITLE
env override new user question

### ISSUE THREAD

**Issue by dtanner** [uchuhimo/konf#51]
I'm having trouble figuring out the correct syntax/configuration for `from.env()` support.
I created a repo with some tests showing the TOML config file I'd like to use.
https://github.com/dtanner/konf-question

Can you help me understand what I need to do to make the test [config from toml and env should produce y for optionA name](https://github.com/dtanner/konf-question/blob/master/src/test/kotlin/question/AppTest.kt#L25) pass?  It's currently failing with this error:
```
com.uchuhimo.konf.source.LoadException: fail to load root

	at com.uchuhimo.konf.source.SourceKt.loadItem(Source.kt:530)
	at com.uchuhimo.konf.BaseConfig.addItem(BaseConfig.kt:542)
	at com.uchuhimo.konf.RequiredConfigProperty.provideDelegate(Config.kt:399)
	at question.AppKt.loadConfigFromTomlAndEnv(App.kt:45)
	at question.AppTest.config from toml and env should produce y for optionA name(AppTest.kt:29)
	....
        com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: com.uchuhimo.konf.source.ObjectMappingException: unable to map source {appconfig={outputs={optiona={name=y}}}} in [type: system-environment] to value of type AppConfig
	at com.uchuhimo.konf.source.SourceKt.toValue(Source.kt:834)
	at com.uchuhimo.konf.source.SourceKt.loadItem(Source.kt:523)
	... 55 more
Caused by: com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException: Instantiation of [simple type, class question.AppConfig] value failed for JSON property outputs due to missing (therefore NULL) value for creator parameter outputs which is a non-nullable type
 at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: question.AppConfig["outputs"])
```

**Comment by chearius** [comment]
**TL;DR: If you use data classes with `from.env()` use lower case field names or replace with ConfigSpec**

The problem is, that `from.env()` enables Konf feature `LOAD_KEYS_CASE_INSENSITIVELY`. So all keys are converted to lower case and the mapper cannot find the correct property in your data class. If you rename `optionA` to `optiona`, everything works.

This seems to happen only in combination with data classes. If you use ConfSpec objects, it works as expected:

```
object AppConfig : ConfigSpec("") {
    object Outputs : ConfigSpec("outputs") {
        val optionA by optional<OptionA?>(default = null)
    }
}

data class OptionA(val name: String)

fun loadConfigFromTomlOnly(configFile: File): Config {
    return Config { addSpec(AppConfig) }
        .from.toml.file(configFile)
}

fun loadConfigFromTomlAndEnv(configFile: File): Config {
    return Config { addSpec(AppConfig)}
        .from.toml.file(configFile)
        .from.env()
}
```

And access options using `val optionA = config[AppConfig.Outputs.optionA]`.

**Comment by dtanner** [comment]
ahh - thanks @chearius !
